src/super_gradients/common/environment/monitoring/monitoring.py

```diff
@@ -0,0 +1,84 @@
+import time
+import threading
+
+from super_gradients.common.environment.env_helpers import multi_process_safe
+from super_gradients.common.environment.monitoring import disk, virtual_memory, network, cpu, gpu
+from super_gradients.common.environment.monitoring.utils import average
+from super_gradients.common.environment.monitoring.data_models import StatAggregator, GPUStatAggregatorIterator
+from torch.utils.tensorboard import SummaryWriter
+
+
+class SystemMonitor:
+    """Monitor and write to tensorboard the system statistics, such as CPU usage, GPU, ...
+
+    :param tensorboard_writer:  Tensorboard object that will be used to save the statistics
+    :param extra_gpu_stats:     Set to True to get extra gpu statistics, such as gpu temperature, power usage, ...
+                                Default set to False, because this reduces the tensorboard readability.
+    """
+
+    def __init__(self, tensorboard_writer: SummaryWriter, extra_gpu_stats: bool = False):
```

Comment:
to make it cleaner, please add a function `close()` that breaks the `while True`
```
while self.running:
   ...

def close():
    self.running = False
```

Reply:
you can call close from the sg_logger of course

Reply:
Added in 00c6dfdfbf3f9325fe515e07dc790ef86c720cfe